File: php/features.php
<?php
declare(strict_types=1);

// ColoradoInformationMarketplace SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class ColoradoInformationMarketplaceFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new ColoradoInformationMarketplaceBaseFeature();
            case "test":
                return new ColoradoInformationMarketplaceTestFeature();
            default:
                return new ColoradoInformationMarketplaceBaseFeature();
        }
    }
}

File: py/features.py
# ColoradoInformationMarketplace SDK feature factory

from feature.base_feature import ColoradoInformationMarketplaceBaseFeature
from feature.test_feature import ColoradoInformationMarketplaceTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ColoradoInformationMarketplaceBaseFeature(),
        "test": lambda: ColoradoInformationMarketplaceTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
